Sources/NIOExtras/PCAPRingBuffer.swift

```diff
@@ -0,0 +1,81 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+// MARK: NIOPCAPRingBuffer
+/// Storage for the most recent set of packets captured subject to constraints.
+/// Use `addFragment` as the sink to a `NIOWritePCAPHandler` and call `emitPCAP`
+/// when you wish to get the recorded data.
+public class NIOPCAPRingBuffer {
```

Comment:
This must note that the API is not thread-safe, and users are required to ensure that they only call it from the correct thread.